src/TestSuite/MiddlewareDispatcher.php

```diff
@@ -74,25 +72,26 @@ class MiddlewareDispatcher
      * @param array|null $constructorArgs The constructor arguments for your application class.
      *   Defaults to `['./config']`
      * @throws \LogicException If it cannot load class for use in integration testing.
-     * @psalm-param \Cake\Core\HttpApplicationInterface::class|\Cake\Core\ConsoleApplicationInterface::class|null $class
+     * @psalm-param class-string<\Cake\Core\HttpApplicationInterface>|null $class
      */
     public function __construct(
         TestCase $test,
         ?string $class = null,
         ?array $constructorArgs = null
     ) {
         $this->_test = $test;
-        $this->_class = $class ?: Configure::read('App.namespace') . '\Application';
+        if ($class === null) {
+            /** @psalm-var class-string<\Cake\Core\HttpApplicationInterface> */
+            $class = Configure::read('App.namespace') . '\Application';
+        }
+        $this->_class = $class;
         $this->_constructorArgs = $constructorArgs ?: [CONFIG];
 
-        try {
-            $reflect = new ReflectionClass($this->_class);
-            /** @var \Cake\Core\HttpApplicationInterface $app */
-            $app = $reflect->newInstanceArgs($this->_constructorArgs);
-            $this->app = $app;
-        } catch (ReflectionException $e) {
-            throw new LogicException("Cannot load `{$this->_class}` for use in integration testing.", 0, $e);
+        if (!class_exists($this->_class)) {
+            throw new LogicException("Cannot load `{$this->_class}` for use in integration testing.", 0);
```

Comment:
This feels like a `RuntimeException` rather than `LogicException` to me but I didn't change it to keep existing test case passing.